app/models/tweet.rb

```diff
@@ -0,0 +1,34 @@
+class Tweet
+  # No need for db in this example, but ActiveModel is useful
+  include ActiveModel::Model
+
+  attr_accessor :user, :content, :id
+  attr_writer :tweeted_at, :hearts, :retweets
+
+  validates_length_of :content, maximum: 280, allow_blank: false
+
+  # Set sensible defaults, unbacked by db
```

Comment:
I was not able to use the `attributes` API here, although I tried.  Even digging into the code, this required adding a bunch of methods to Tweet that felt unnecessary for this little demo.